apps/site/assets/js/support-form.js

```diff
@@ -155,31 +155,45 @@ export function handleUploadedPhoto($, file, $container, toUpload) {
   $container.append(preview);
 }
 
-export function resizeImage(file) {
-  return new Promise((resolve, reject) => {
+const resizeImage = (file) =>
+  new Promise((resolve, reject) => {
     const fr = new FileReader();
     fr.onerror = () => {
       fr.abort();
       reject(new DOMException("Error parsing file."));
     };
-    fr.onload = function() {
+    fr.onload = () => {
       const img = new Image();
-      img.onload = function() {
+      img.onload = () => {
         const dim = rescale({ width: img.width, height: img.height });
         const canvas = document.createElement("canvas");
         canvas.width = dim.width;
         canvas.height = dim.height;
         const ctx = canvas.getContext("2d");
         ctx.drawImage(img, 0, 0, dim.width, dim.height);
-        canvas.toBlob(function(blob) {
-          resolve(blob);
-        }, "image/jpeg");
+        let isToBlobSupported = false;
+        try {
+          canvas.toBlob(blob => {
+            resolve(blob);
+          }, "image/jpeg");
+          isToBlobSupported = true;
+        } catch (e) {
+          // toBlob is not supported. Will next try msToBlob.
```

Comment:
Should I still run another catch if msToBlob fails too?  So then it would be a nested try-catch?

Reply:
If `msToBlob` works for all browsers, you'd just have to use that one instead of choosing between the two? (we'd have to double check that)
If not, what about
```
    if (canvas.msToBlob) {
        // IE
    } else {
		…
    }
```
instead of nesting?

Reply:
Aha!  This is golden, thanks.  `msToBlob` is unique to IE, so the if-else you shared works nicely.  It didn't occur to me that I can use `canvas.msToBlob` in an existence check.  Nice!